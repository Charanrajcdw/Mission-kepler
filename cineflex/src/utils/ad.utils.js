import ad1 from "../assets/advertisements/small/small-ad1.png";
import ad2 from "../assets/advertisements/small/small-ad2.png";

const ads = [ad1, ad2];

export const getRandomAd = () => {
  return ads[Math.floor(Math.random() * ads.length)];
};
