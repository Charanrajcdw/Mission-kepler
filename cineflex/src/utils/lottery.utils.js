import { LOTTERY } from "../constants";

export const checkLotteryNo = (enteredNumber) => {
  const number = +enteredNumber;
  if (isNaN(number) || enteredNumber.length != 10) return LOTTERY.invalidNumber;
  const lastDigit = +enteredNumber.slice(-1);
  if (lastDigit % 2 !== 0) return LOTTERY.unluckyMessage;
};
