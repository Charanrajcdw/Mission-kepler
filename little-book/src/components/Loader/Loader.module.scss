.loader {
  position: relative;
  top: 50%;
  left: 50%;
  transform: translate(-15%, -40%);
}
