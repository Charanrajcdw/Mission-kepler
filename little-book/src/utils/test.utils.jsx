import { configureStore } from "@reduxjs/toolkit";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import ThemeContextProvider from "../contexts/theme.context";
import blogSlice from "../store/slices/blogSlice";
import { THEME } from "../constants";

const customRender = (
  ui,
  {
    preloadedState = {},
    store = configureStore({
      reducer: { blogs: blogSlice.reducer },
      preloadedState,
    }),
    theme = THEME.light,
    ...options
  } = {}
) => {
  const ProviderWrapper = ({ children }) => {
    return (
      <ThemeContextProvider theme={theme}>
        <Provider store={store}>{children}</Provider>
      </ThemeContextProvider>
    );
  };
  return render(ui, { wrapper: ProviderWrapper, ...options });
};

const customCreateSnapshot = (
  ui,
  {
    preloadedState = {},
    store = configureStore({
      reducer: { blogs: blogSlice.reducer },
      preloadedState,
    }),
    theme = THEME.light,
  } = {}
) => {
  const ProviderWrapper = ({ children }) => {
    return (
      <ThemeContextProvider theme={theme}>
        <Provider store={store}>{children}</Provider>
      </ThemeContextProvider>
    );
  };
  return renderer.create(<ProviderWrapper>{ui}</ProviderWrapper>).toJSON();
};

export * from "@testing-library/react";
export { default as userEvent } from "@testing-library/user-event";
export { customRender as render, customCreateSnapshot as createSnapshot };
