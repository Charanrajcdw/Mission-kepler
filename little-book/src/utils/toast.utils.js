import { toast } from "react-toastify";

export const showSuccessToast = (message, currentTheme) => {
  toast.success(message, {
    position: "top-right",
    autoClose: 2000,
    closeOnClick: true,
    pauseOnHover: true,
    draggable: true,
    theme: currentTheme,
  });
};

export const showWarningToast = (message, currentTheme) => {
  toast.warn(message, {
    position: "top-right",
    autoClose: 2000,
    closeOnClick: true,
    pauseOnHover: true,
    draggable: true,
    theme: currentTheme,
  });
};
