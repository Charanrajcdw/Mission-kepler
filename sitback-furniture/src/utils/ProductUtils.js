export const transformToIndianRupee = (price) => {
  price = parseInt(price);
  return price.toLocaleString("en-IN");
};

export const getGuaranteeMessage = (guarantee) => {
  return guarantee === 1 ? `1 YEAR GUARANTEE` : `${guarantee} YEARS GUARANTEE`;
};
