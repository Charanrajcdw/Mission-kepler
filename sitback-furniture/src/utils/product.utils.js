import { CART } from "../constants/constants";

export const transformToIndianRupee = (price) => {
  price = +price;
  return price ? price.toLocaleString("en-IN", { maximumFractionDigits: 2 }) : 0;
};

export const getGuaranteeMessage = (guarantee) => {
  return guarantee === 1 ? `1 YEAR GUARANTEE` : `${guarantee} YEARS GUARANTEE`;
};

export const calculateTotalPrice = (items) => {
  const price = items.reduce((total, product) => total + parseInt(product.price) * product.quantity, 0);
  return transformToIndianRupee(price);
};

export const getItemsFromLocalStorage = (key) => {
  return JSON.parse(localStorage.getItem(key)) ?? [];
};

export const setItemsToLocalStorage = (key, items) => {
  localStorage.setItem(key, JSON.stringify(items));
};

export const addToCart = (product, quantity) => {
  const cartItems = getItemsFromLocalStorage(CART.cart);
  const productIndex = cartItems.findIndex((item) => item.id === product.id);
  if (productIndex >= 0) {
    cartItems[productIndex].quantity += quantity;
    if (cartItems[productIndex].quantity === 0) {
      cartItems.splice(productIndex, 1);
    }
  } else {
    cartItems.push({ ...product, quantity: quantity });
  }
  setItemsToLocalStorage(CART.cart, cartItems);
  return cartItems;
};

export const addToWishlist = (product) => {
  const wishlistItems = getItemsFromLocalStorage(CART.wishlist);
  const productIndex = wishlistItems.findIndex((item) => item.id === product.id);
  if (productIndex < 0) {
    wishlistItems.push(product);
  }
  setItemsToLocalStorage(CART.wishlist, wishlistItems);
  return wishlistItems;
};

export const removeFromWishlist = (product) => {
  const wishlistItems = getItemsFromLocalStorage(CART.wishlist);
  const newWishlistItems = wishlistItems.filter((item) => item.id !== product.id);
  setItemsToLocalStorage(CART.wishlist, newWishlistItems);
  return newWishlistItems;
};
