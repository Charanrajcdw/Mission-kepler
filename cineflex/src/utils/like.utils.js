import { LIKES } from "../constants";
import { localStorageHelper } from "./localStorage.utils";

const { get, set } = localStorageHelper;

export const storeLikedMovie = (id) => {
  const likedMovies = get(LIKES.key) ?? [];
  if (!likedMovies.includes(id)) likedMovies.push(id);
  set(LIKES.key, likedMovies);
};
