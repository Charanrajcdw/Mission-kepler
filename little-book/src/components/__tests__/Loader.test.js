import { createSnapshot } from "../../utils/test.utils";
import Loader from "../Loader/Loader";

it("should match loader snapshot", () => {
  const loader = createSnapshot(<Loader />);
  expect(loader).toMatchSnapshot();
});
