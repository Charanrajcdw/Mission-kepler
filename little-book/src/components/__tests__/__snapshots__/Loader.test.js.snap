// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match loader snapshot 1`] = `
<div
  className="loader"
>
  <span
    className="loader"
    style={
      {
        "display": "inherit",
        "fontSize": "0",
        "height": "204px",
        "left": 68,
        "position": "relative",
        "top": 68,
        "width": "204px",
      }
    }
  >
    <span
      style={
        {
          "animation": "react-spinners-FadeLoader-fade 1.2s 0.12s infinite ease-in-out",
          "animationFillMode": "both",
          "backgroundColor": "#bbb",
          "borderRadius": "10px",
          "height": "50px",
          "left": "0",
          "margin": "50px",
          "position": "absolute",
          "top": "68px",
          "transition": "2s",
          "width": "10px",
        }
      }
    />
    <span
      style={
        {
          "animation": "react-spinners-FadeLoader-fade 1.2s 0.24s infinite ease-in-out",
          "animationFillMode": "both",
          "backgroundColor": "#bbb",
          "borderRadius": "10px",
          "height": "50px",
          "left": "46.36363636363636px",
          "margin": "50px",
          "position": "absolute",
          "top": "46.36363636363636px",
          "transform": "rotate(-45deg)",
          "transition": "2s",
          "width": "10px",
        }
      }
    />
    <span
      style={
        {
          "animation": "react-spinners-FadeLoader-fade 1.2s 0.36s infinite ease-in-out",
          "animationFillMode": "both",
          "backgroundColor": "#bbb",
          "borderRadius": "10px",
          "height": "50px",
          "left": "68px",
          "margin": "50px",
          "position": "absolute",
          "top": "0",
          "transform": "rotate(90deg)",
          "transition": "2s",
          "width": "10px",
        }
      }
    />
    <span
      style={
        {
          "animation": "react-spinners-FadeLoader-fade 1.2s 0.48s infinite ease-in-out",
          "animationFillMode": "both",
          "backgroundColor": "#bbb",
          "borderRadius": "10px",
          "height": "50px",
          "left": "46.36363636363636px",
          "margin": "50px",
          "position": "absolute",
          "top": "-46.36363636363636px",
          "transform": "rotate(45deg)",
          "transition": "2s",
          "width": "10px",
        }
      }
    />
    <span
      style={
        {
          "animation": "react-spinners-FadeLoader-fade 1.2s 0.6s infinite ease-in-out",
          "animationFillMode": "both",
          "backgroundColor": "#bbb",
          "borderRadius": "10px",
          "height": "50px",
          "left": "0",
          "margin": "50px",
          "position": "absolute",
          "top": "-68px",
          "transition": "2s",
          "width": "10px",
        }
      }
    />
    <span
      style={
        {
          "animation": "react-spinners-FadeLoader-fade 1.2s 0.72s infinite ease-in-out",
          "animationFillMode": "both",
          "backgroundColor": "#bbb",
          "borderRadius": "10px",
          "height": "50px",
          "left": "-46.36363636363636px",
          "margin": "50px",
          "position": "absolute",
          "top": "-46.36363636363636px",
          "transform": "rotate(-45deg)",
          "transition": "2s",
          "width": "10px",
        }
      }
    />
    <span
      style={
        {
          "animation": "react-spinners-FadeLoader-fade 1.2s 0.84s infinite ease-in-out",
          "animationFillMode": "both",
          "backgroundColor": "#bbb",
          "borderRadius": "10px",
          "height": "50px",
          "left": "-68px",
          "margin": "50px",
          "position": "absolute",
          "top": "0",
          "transform": "rotate(90deg)",
          "transition": "2s",
          "width": "10px",
        }
      }
    />
    <span
      style={
        {
          "animation": "react-spinners-FadeLoader-fade 1.2s 0.96s infinite ease-in-out",
          "animationFillMode": "both",
          "backgroundColor": "#bbb",
          "borderRadius": "10px",
          "height": "50px",
          "left": "-46.36363636363636px",
          "margin": "50px",
          "position": "absolute",
          "top": "46.36363636363636px",
          "transform": "rotate(45deg)",
          "transition": "2s",
          "width": "10px",
        }
      }
    />
  </span>
</div>
`;
